src/DataWrangling/JRA55.jl

```diff
@@ -159,64 +204,266 @@ function jra55_field_time_series(variable_name;
-    if Nt <= chunk_size # one chunk will do
-        fill_halo_regions!(fts)
-    else # need multiple chunks
-        start = 1
-        while start < Nt
-            stop = min(Nt, start + chunk_size - 1)
-            fts_chunk = Tuple(fts[n] for n = start:stop)
-            fill_halo_regions!(fts_chunk)
-            start += chunk_size
+    fill_halo_regions!(native_fts)
+
+    if isnothing(grid)
+        return native_fts
+    else # make a new FieldTimeSeries and interpolate native data onto it.
+        boundary_conditions = FieldBoundaryConditions(grid, (LX, LY, Nothing))
+        fts = FieldTimeSeries{LX, LY, Nothing}(grid, times; boundary_conditions)
+
+        interpolate!(fts, native_fts)
+
+        return fts
+    end
+end
+
+# TODO: check also the time indices
+function retrieve_and_maybe_write_jra55_data(chunks, grid, times, loc, boundary_conditions, data, jra55_native_grid; 
+                                             interpolated_file = nothing, 
+                                             shortname = nothing)
+
+    if !isfile(interpolated_file) # File does not exist, let's rewrite it
+        
+        @info "rewriting the jra55 data into an Oceananigans compatible format"
+        interpolate_and_write_timeseries!(data, loc, grid, times, interpolated_file, shortname, boundary_conditions, jra55_native_grid)
+    end
+
+    file = jldopen(interpolated_file)
+
+    if file["serialized/grid"] != grid # File exists but the data is on another grid, remove it and rewrite it
+
+        close(file)
+        @info "the saved boundary data is on another grid, deleting the old boundary file"
+        rm(interpolated_file; force=true)
+        
+        @info "rewriting the jra55 data into an Oceananigans compatible format"
+        interpolate_and_write_timeseries!(data, loc, grid, times, interpolated_file, shortname, boundary_conditions, jra55_native_grid)
+
+    else # File exists and the data is on the correct grid
+
+        if !(shortname ∈ keys(file["timeseries"])) # `shortname` is not in the file
+            close(file)
+
+            @info "rewriting the jra55 data into an Oceananigans compatible format"
+            interpolate_and_write_timeseries!(data, loc, grid, times, interpolated_file, shortname, boundary_conditions, jra55_native_grid)
         end
+
+        # File is there and `shortname` is in the file (probably the time indices are not correct)
+        # TODO: check the time range includes the time range of the simulation
+        # check_time_indices!(args...)
     end
 
+    fts = FieldTimeSeries(interpolated_file, shortname; backend = InMemory(; chunk_size = chunks))
+
     return fts
 end
 
+"""
+    interpolate_and_write_timeseries!(data, loc, grid, times, path, name, bcs, native_grid)
+
+Interpolates and writes a time series of `data` at `times` onto disk in an Oceananigans compatible format.
+"""
+function interpolate_and_write_timeseries!(data, loc, grid, times, path, name, bcs, native_grid)
```

Comment:
This function is not specific to JRA55 so I don't think it belongs here in JRA55.jl